process_nwb/wavelet_transform.py

```diff
@@ -358,7 +360,7 @@ def store_wavelet_transform(elec_series, processing, filters='rat', hg_only=True
                                                    metric='amplitude',
                                                    source_timeseries=elec_series,
                                                    starting_time=elec_series.starting_time,
-                                                   rate=rate,
+                                                   rate=post_resample_rate,
```

Comment:
I think this will not work if `post_resample_rate` is `None`. having something like this before the DecompositionSeries is created and leaving `rate` here should work
```python
if post_resample_rate is not None:
    rate = post_resample_rate
```

Reply:
It looks like the original `rate` is used on line 373 so overwriting it result in incorrectly resampled data. What about this instead:

if post_resample_rate is None:
    post_resample_rate = rate

Reply:
373 is in the non-chunked condition, so only this path or that would would happen, not both. But what you wrote would also work.